source/Nuke.Common/Tools/OctoVersion/OctoVersionAttribute.cs

```diff
@@ -22,27 +24,232 @@ namespace Nuke.Common.Tools.OctoVersion
+        /// Supports AzurePipelines, TeamCity, AppVeyor and GitHubActions
+        /// </summary>
+        public bool UpdateBuildNumber { get; set; } = NukeBuild.IsServerBuild;
+
+        /// <summary>
+        /// Automatically detect the branch to pass to OctoVersion, based on the git worktree
+        /// </summary>
+        public bool AutoDetectBranch { get { return _autoDetectBranch.GetValueOrDefault(); } set { _autoDetectBranch = value; } }
+        private bool? _autoDetectBranch;
+
+        /// <summary>
+        /// The name of a Nuke parameter specifying whether to auto detect the branch to pass to Octoversion, based on the git worktree
+        /// </summary>
+        [CanBeNull] public string AutoDetectBranchParameter { get; set; }
+
+        /// <summary>
+        /// The branch to pass to OctoVersion
+        /// </summary>
+        [CanBeNull] public string Branch { get; set; }
+        
+        /// <summary>
+        /// The name of a Nuke parameter containing the branch to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string BranchParameter { get; set; }
+
+        /// <summary>
+        /// The Major version number to pass to OctoVersion
+        /// </summary>
+        [CanBeNull] public int Major { get { return _major.GetValueOrDefault(); } set { _major = value; } }
+        private int? _major;
+
+        /// <summary>
+        /// The name of a Nuke parameter containing the Major version number to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string MajorParameter { get; set; }
+
+        /// <summary>
+        /// The Minor version number to pass to OctoVersion
+        /// </summary>
+        [CanBeNull] public int Minor { get { return _minor.GetValueOrDefault(); } set { _minor = value; } }
+        private int? _minor;
+
+        /// <summary>
+        /// The name of a Nuke parameter containing the Minor version number to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string MinorParameter { get; set; }
+
+        /// <summary>
+        /// The Patch version number to pass to OctoVersion
+        /// </summary>
+        [CanBeNull] public int Patch { get { return _patch.GetValueOrDefault(); } set { _patch = value; } }
+        private int? _patch;
+
+        /// <summary>
+        /// The name of a Nuke parameter containing the Patch version number to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string PatchParameter { get; set; }
 
         public override object GetValue(MemberInfo member, object instance)
         {
+            // Wrap log output in a NUKE Section
+            using (Logger.Block("[OctoVersion] Injection"))
+            {
+                LogAttributeProperties();
+
+                return GetValueInternal(member, instance);
+            }
+        }
+
+        private void LogAttributeProperties()
```

Comment:
Not a fan of this for two reasons:
1. We don't have log output like this anywhere else
2. There is an `AuditBuildMembers` attribute in `Nuke.Enterprise` which logs all the members, but it could be extended to also log the configuration of such attributes.

Reply:
Keep in mind it is trace level logging so only shows up when running with verbosity enabled.  It was useful to me when troubleshooting/consuming the attribute to understand what was being passed in.  The consumers of this will be looking at a build log in a CI tool like teamcity or github actions etc so it seemed like any information we can output as to what is going on would be helpful

The reason for attempting to put it in a LogBlock section is, that the attribute injection occurs BEFORE any build targets and having all of the logged output from the setting up to run the tool AND the actual tooll execution itself, floating on it's own before any sectioned build targets seemed a bit off putting.  Putting it in a log block means that it is expanded/collapsed in the supported CI platforms and much more clearly delineated that it is all of the relevant log output from the [OctoVersion] attribute injection.  
It also defaults to collapsed in GHActions...
so we end up with this:
![image](https://user-images.githubusercontent.com/5425163/131432466-fcfc4f5a-2e2b-4b15-83b2-8bfeebdc064e.png)

the alternative would be we have quite a bit of log output (even if we remove the parameter logging i added here, just from the tool execution) that isnt collapsed and seems more cluttered.

if you implemented support for SubBlocks im not sure it would look much different here, as we arent in a block yet anyway (we are in the attribute injection phase, before any build targets are running).

I felt this was cleaner but if you want it to be removed then we can do that

Reply:
Ive removed the parameter logging but i really would like to keep the log block as it contains the output from the tool nicely... otherwise these value injected attributes are just floating off in no man's land

Reply:
I understand the argument, but logging is a part that I'd like to come back to a bit later. In the meantime, I don't want to introduce any tracing or logging that could be solved in a more general and configurable way, as mentioned with the `AuditBuildMembers` attribute.

Reply:
LogBlock now removed.  It would definitely be nice to see Nuke core do something here so that the attribute injection stuff is a little more clear and bounded, in the log outputs.  It's not a blocker for this PR though, so ive removed the LogBlock